git_machete/tests/functional/test_machete.py

```diff
@@ -1670,10 +1673,10 @@ def test_anno_prs(self) -> None:
     git_api_state_for_test_create_pr = MockGithubAPIState([{'head': {'ref': 'ignore-trailing'}, 'user': {'login': 'github_user'}, 'base': {'ref': 'hotfix/add-trigger'}, 'number': '3', 'html_url': 'www.github.com'}])
 
     @mock.patch('git_machete.options.CommandLineOptions', FakeCommandLineOptions)
+    @mock.patch('git_machete.github.GITHUB_REMOTE_PATTERNS', FAKE_GITHUB_REMOTE_PATTERNS)
```

Comment:
Here, `FAKE_GITHUB_REMOTE_PATTERNS` was added because somehow, `Remotes are defined for this repository, but none of them corresponds to GitHub` error is still raised. Same logic is used in `test_retarget_pr` and `test_anno_prs`, where solution works as expected.

Reply:
Whoops okay this error might be specific not just to the tests, but also happen in the actual (production) use. Let's not ignore it... pls:
1. remove this `@mock.patch`
2. add some extra information to the `Remotes are ...` error message, like e.g. an output of `git remotes -v` for instance; it's even okay if we release v3.4.0 with that extended error message

Reply:
Although... wait, isn't the problem related to the `git push` that happens before a PR is opened? pls investigate 🕵🏻

Reply:
If so, then mocking `GITHUB_REMOTE_PATTERNS` is probably a reasonable idea, same as in the test for `checkout-prs`

Reply:
Problem is very tricky. Unfortunately it's not connected to `push` operation because error is raised before, while program determines what is correct remote for github, at `__derive_remote_and_github_org_and_repo`. Since error occurs always at same line which points to line 1815 in test, where `create_github_pr` is called directly from `MacheteClient` instance, not from `launch` command. During failures `__remotes_cached` holds only remote but `git remote -v` outputs two as it's supposed to be. 

I tried to catch it on my machine , with same  test repository setup and with direct calls to `create_github_pr`, but  on 1000 tries every single one was correct. Same story with running tox locally. I also tried to `flush_caches` right before `__derive_remote_and_github_org_and_repo` to enforce reading remotes again, but it doesn't solve the problem. Adding delays does not do the job too.

Reply:
finally clearing out `__remotes_cached` with `flush_caches` fixes the problem. 

But it's still unknown, why it happens so randomly, and only in CI environment